xdsl/dialects/experimental/aie.py

```diff
@@ -0,0 +1,1129 @@
+class PLIOOp(IRDLOperation):
+    name = "AIE.plio"
+
+    col = attr_def(IntegerAttr[IntegerType])
+
+    def __init__(self, col: IntegerAttr[IntegerType]):
+        super().__init__(attributes={"col": col}, result_types=[IndexType()])
+
+
+@irdl_op_definition
+class PacketDestOp(IRDLOperation):
+    name = "AIE.packet_dest"
+
+    bundle = attr_def(WireBundleAttr)
+    channel = attr_def(IntegerAttr[IntegerType])
+
+    tile = operand_def(IndexType())
+
+    def __init__(
+        self,
+        bundle: WireBundleAttr,
+        channel: IntegerAttr[IntegerType],
+        tile: Operation | SSAValue,
+    ):
+        super().__init__(
+            attributes={"bundle": bundle, "channel": channel}, operands=[tile]
+        )
+
+    def print(self, printer: Printer):
+        printer.print(
+            "<", self.tile, ", ", self.bundle.data, " : ", self.channel.value.data, ">"
+        )
+
+
+@irdl_op_definition
+class PacketFlowOp(IRDLOperation):
+    name = "AIE.packet_flow"
+
+    ID = attr_def(IntegerAttr[IntegerType])
+    region = region_def()
+
+    def __init__(self, ID: IntegerAttr[IntegerType], region: Region):
+        super().__init__(attributes={"ID": ID}, regions=[region])
+
+    def print(self, printer: Printer):
+        printer.print("(", f"0x{self.ID.value.data:X}", ") ")
+        printer.print_region(self.region)
+
+
+@irdl_op_definition
+class PacketRuleOp(IRDLOperation):
+    name = "AIE.rule"
+
+    mask = attr_def(IntegerAttr[IntegerType])
+    value = attr_def(IntegerAttr[IntegerType])
+
+    amsel = operand_def(IndexType())
+
+    def __init__(
+        self,
+        mask: IntegerAttr[IntegerType],
+        value: IntegerAttr[IntegerType],
+        amsel: Operation | SSAValue,
+    ):
+        super().__init__(attributes={"mask": mask, "value": value}, operands=[amsel])
+
+
+@irdl_op_definition
+class PacketRulesOp(IRDLOperation):
+    name = "AIE.packetrules"
+
+    sourceBundle = attr_def(WireBundleAttr)
+    sourceChannel = attr_def(IntegerAttr[IntegerType])
+
+    def __init__(
+        self, sourceBundle: WireBundleAttr, sourceChannel: IntegerAttr[IntegerType]
+    ):
+        super().__init__(
+            attributes={"sourceBundle": sourceBundle, "sourceChannel": sourceChannel}
+        )
+
+
+@irdl_op_definition
+class PacketSourceOp(IRDLOperation):
+    name = "AIE.packet_source"
+
+    bundle = attr_def(WireBundleAttr)
+    channel = attr_def(IntegerAttr[IntegerType])
+    tile = operand_def(IndexType())
+
+    def __init__(
+        self,
+        bundle: WireBundleAttr,
+        channel: IntegerAttr[IntegerType],
+        tile: Operation | SSAValue,
+    ):
+        super().__init__(
+            attributes={"bundle": bundle, "channel": channel}, operands=[tile]
+        )
+
+    def print(self, printer: Printer):
+        printer.print(
+            "<", self.tile, ", ", self.bundle.data, " : ", self.channel.value.data, ">"
+        )
+
+
+@irdl_op_definition
+class putCascade(IRDLOperation):
```

Comment:
```suggestion
class PutCascade(IRDLOperation):
```
Please make all the classes PascalCase